### PR TITLE
Add proposal for verifying a custom domain

### PR DESCRIPTION
**Pull request by maleck13**
proposal for #80

### CODE REVIEW

File: docs/proposals/custom-domain-support.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+We assume individuals within a single workspace are considered trusted. So proving ownership of a domain will happen at the workspace level.
+To prove ownership of a domain we will use DNS validation via a txt record. This is the same process used for things like lets encrypt and GH pages. The assumption is that if you can make a change to the DNS Zone of a given domain, then you have control over where that domain resolves and so are considered the owner or at least authorised to use the domain.
+ 
+
+Flow:
+
+- End user creates an Ingress resource with a custom domain in the host field 
+- Mutating webhook sees custom domain 
+- Mutating webhook checks for a DomainVerification resource (outlined below), if not present it will create one. If present and not validated, it will strip out the custom domain and any associated tls section and adds them to an annotation for later reference.
```
Comment (mikenairn) [inline docs/proposals/custom-domain-support.md:43]: I'm still not sure if it should do this tbh. Are we not likely to end up with lots of unnecessary `DomainVerification` resources, one added every time someone create an ingress with some host i.e. xyz.nip.io, presumably always waiting for something to happen, but will never become verified.

Is a better approach to not just expect someone to create the resource when they know they want to verify a domain?

Comment (mikenairn) [inline docs/proposals/custom-domain-support.md:43]: I'm still not sure if it should do this tbh. Are we not likely to end up with lots of unnecessary `DomainVerification` resources, one added every time someone create an ingress with some host i.e. xyz.nip.io, presumably always waiting for something to happen, but will never become verified.

Is a better approach to not just expect someone to create the resource when they know they want to verify a domain?

**Comment by maleck13** [comment]
@mikenairn my comments seem to be in a pending state so not sure if you have seen it: Will add here to be sure:

> Is a better approach to not just expect someone to create the resource when they know they want to verify a domain?

 I think the question here is how should the user express their intent to use a custom domain.

By adding it to their ingress
By adding it to a separate resource
If we say the intent is expressed by adding it to their Ingress then we need to auto create the resource. The number of resources should be reduced because a CustomDomainVerification resource for app.com will cover any subdomain of that also. When no ingresses are present for a customdomainverification it should be removed.

File: docs/proposals/custom-domain-support.md (hunk 1 above)
Comment (mikenairn) [reply docs/proposals/custom-domain-support.md]: Spoke offline about this and I'm happy that we can create the verification resource and keep the ingress the resource users interact with assuming appropriate status can be added there relating to the state of the custom domain verification.

**Review by mikenairn — APPROVED** [review]

**Pull request merged**